onnx/backend/test/cpp/driver_test.cc

```diff
@@ -0,0 +1,140 @@
+                std::vector<onnxTensorDescriptorV1> input_descriptor,
+                    output_descriptor, result_descriptor;
+                for (auto input : proto_test_data.inputs_) {
+                  input_descriptor.push_back(
+                      ProtoToOnnxTensorDescriptor(input));
+                }
+                for (auto output : proto_test_data.outputs_) {
+                  output_descriptor.push_back(
+                      ProtoToOnnxTensorDescriptor(output));
+                  onnxTensorDescriptorV1 result;
+                  result.tag = ONNXIFI_TAG_TENSOR_DESCRIPTOR_V1;
+                  result.name = "result";
+                  result.dataType = output.data_type();
+                  result.memoryType = ONNXIFI_MEMORY_TYPE_CPU;
+                  result.dimensions = output.dims().size();
+                  result.shape = (unsigned long long*)output.dims().data();
+                  result.buffer =
+                      (onnxPointer) new char[sizeof(output.raw_data().size())];
+                  result_descriptor.push_back(result);
+                }
+                EXPECT_EQ(
+                    lib.onnxSetGraphIO(
+                        graph,
+                        input_descriptor.size(),
+                        input_descriptor.data(),
+                        result_descriptor.size(),
+                        result_descriptor.data()),
+                    ONNXIFI_STATUS_SUCCESS);
+
+                onnxMemoryFenceV1 inputFence, outputFence;
+                inputFence.tag = ONNXIFI_TAG_MEMORY_FENCE_V1;
+                outputFence.tag = ONNXIFI_TAG_MEMORY_FENCE_V1;
+                inputFence.type = ONNXIFI_SYNCHRONIZATION_EVENT;
+                outputFence.type = ONNXIFI_SYNCHRONIZATION_EVENT;
+
+                // EXPECT_EQ(lib.onnxInitEvent(backend, inputFence),
+                // ONNXIFI_STATUS_SUCCESS);
+                EXPECT_EQ(
+                    lib.onnxRunGraph(graph, &inputFence, &outputFence),
+                    ONNXIFI_STATUS_SUCCESS);
+                EXPECT_EQ(
+                    lib.onnxWaitEvent(outputFence.event),
+                    ONNXIFI_STATUS_SUCCESS);
+                for (int i = 0; i < output_descriptor.size(); i++) {
+                  auto output_size = getDescriptorSize(&output_descriptor[i]);
+                  for (int j = 0; j < output_size; j++) {
+                    // size might be a problem!
+                    EXPECT_EQ(
+                        ((char*)output_descriptor[i].buffer)[j],
+                        ((char*)result_descriptor[i].buffer)[j]);
+                  }
+                }
+              }
+              EXPECT_EQ(lib.onnxReleaseGraph(graph), ONNXIFI_STATUS_SUCCESS);
+            }
+        }
+};
+TEST_P(ONNXCppDriverTest, ONNXCppDriverUnitTest){
+	onnxifi_library lib;
+	onnxBackendID backendID;
+	onnxBackend backend;
+        // using default onnxifi backend
+        if (ONNXIFI_BACKEND_USED) {
```

Comment:
Is this used to distinguish our dummy backend and the other real backends?

Reply:
By now I only check the model and the tensor as we dont have real backends, since loading dummy backend here is useless.

Reply:
Got it. In Python land test cases can throw a special exception to indicate skipping a test at *runtime*. Just found gtest doesn't it yet. There is an open pr https://github.com/google/googletest/pull/1544 to add it to gtest, would be good to use it once it's merged to avoid special logic for the dummy backend.